e2e/indexer.e2e-spec.ts

```diff
@@ -26,45 +26,58 @@ describe('Indexer', () => {
-        const p2wkhOutputs = walletHelper.generateAddresses(6, 'p2wpkh');
-        const utxos: UTXO[] = [];
+    async function testSilentBlock(
+        addressType: 'p2wpkh' | 'p2pkh' | 'p2sh-p2wpkh',
+    ) {
+        it(`${addressType} - should ensure that the correct silent block is fetched`, async () => {
+            const taprootOutput = walletHelper.generateAddresses(1, 'p2tr')[0];
+            const outputs = walletHelper.generateAddresses(6, addressType);
+            const utxos: UTXO[] = [];
 
-        for (const output of p2wkhOutputs) {
-            const utxo = await walletHelper.addAmountToAddress(output, 1);
-            utxos.push(utxo);
-        }
+            for (const [index, output] of outputs.entries()) {
+                const utxo = await walletHelper.addAmountToAddress(
+                    output,
+                    1,
+                    addressType,
+                    index,
+                );
+                utxos.push(utxo);
+            }
 
-        const [transaction, txid, blockHash] =
-            await walletHelper.craftAndSpendTransaction(
-                utxos,
-                taprootOutput,
-                5.999,
-                0.001,
+            const [transaction, txid, blockHash] =
+                await walletHelper.craftAndSpendTransaction(
+                    utxos,
+                    taprootOutput,
+                    5.999,
+                    0.001,
+                );
+
+            const transformedTransaction = transformTransaction(
+                transaction,
+                txid,
+                blockHash,
+                walletHelper.currentBlockCount,
+                outputs,
+                indexerService,
             );
 
-        const transformedTransaction = transformTransaction(
-            transaction,
-            txid,
-            blockHash,
-            walletHelper.currentBlockCount,
-            p2wkhOutputs,
-            indexerService,
-        );
+            const silentBlock = silentBlockService.encodeSilentBlock([
+                transformedTransaction,
+            ]);
 
-        const silentBlock = silentBlockService.encodeSilentBlock([
-            transformedTransaction,
-        ]);
+            await new Promise((resolve) => setTimeout(resolve, 15000));
+            const response = await apiHelper.get(
+                `/silent-block/hash/${blockHash}`,
+                {
+                    responseType: 'arraybuffer',
+                },
+            );
 
-        await new Promise((resolve) => setTimeout(resolve, 15000));
-        const response = await apiHelper.get(
-            `/silent-block/hash/${blockHash}`,
-            {
-                responseType: 'arraybuffer',
-            },
-        );
+            expect(response.data as Buffer).toEqual(silentBlock);
+        });
+    }
 
-        expect(response.data as Buffer).toEqual(silentBlock);
-    });
```

Comment:
I would prefer it if the test function, in which i mean` it(`${addressType} - should ensure that the correct silent block is fetched`` is not enclosed in a normal function. It makes the code more readable.